tests/testsuite/cargo_add/overwrite_optional_with_optional/mod.rs

```diff
@@ -0,0 +1,38 @@
+use cargo_test_support::compare::assert_ui;
+use cargo_test_support::prelude::*;
+use cargo_test_support::Project;
+
+use cargo_test_support::curr_dir;
+
+#[cargo_test]
+fn case() {
+    cargo_test_support::registry::init();
+    for name in ["my-package1", "my-package2"] {
+        for ver in [
+            "0.1.1+my-package",
+            "0.2.0+my-package",
+            "0.2.3+my-package",
+            "0.4.1+my-package",
+            "20.0.0+my-package",
+            "99999.0.0+my-package",
+            "99999.0.0-alpha.1+my-package",
+        ] {
+            cargo_test_support::registry::Package::new(name, ver).publish();
+        }
+    }
+
+    let project = Project::from_template(curr_dir!().join("in"));
+    let project_root = project.root();
+    let cwd = &project_root;
+
+    snapbox::cmd::Command::cargo_ui()
+        .arg("add")
+        .arg_line("my-package1 my-package2@0.4.1 --optional")
```

Comment:
We likely don't need two packages and all of those versions of the packages for what this test is doing.

Yes, you copied it from other tests but we want to work to minimize those.  Updating everything can be slow but we can at least not add more to that backlog by ensuring new tests don't do it.

Reply:
Yes, redundant packages are deleted.